packages/web-lib/src/persistence/strategies/local-storage.strategy.ts

```diff
@@ -0,0 +1,69 @@
+import { IStorageStrategy, OperationResult } from '../web-store.types';
+
+export class LocalStorageStrategy implements IStorageStrategy {
+  storage: Storage;
+
+  constructor(storage?: Storage) {
+    this.storage = storage ?? localStorage;
+  }
+
+  async init(): Promise<void> {}
+
+  async getOneByKey<TData>(key: string): Promise<OperationResult<TData>> {
+    try {
+      const storedItem = this.storage.getItem(key);
+      if (!storedItem) {
+        return { success: false, data: 'nothing' as TData };
+      }
+      return {
+        success: true,
+        data: JSON.parse(storedItem) as TData,
+      };
+    } catch (error) {
+      const errorMsg = 'Error in local storage getOneByKey: ';
+      if (error instanceof SyntaxError) {
+        console.error(errorMsg + error.name);
+      } else if (error instanceof Error) {
+        console.error(errorMsg + error.message);
+      }
+      return { success: false, data: 'error' as TData };
+    }
+  }
+
+  async setOneByKey(key: string, value: unknown): Promise<OperationResult<void>> {
+    try {
+      if (await this.storageAvailable()) {
```

Comment:
this is a nice implementation 😊  
a better implementation will be to listen for 'QUOTA_EXCEEDED_ERR' error 
https://stackoverflow.com/questions/13567509/what-happens-when-localstorage-is-full